cython_modules/cython_parse.pyx

```diff
@@ -201,12 +201,35 @@ def decode_byte_size(byte_format_specification):
         raise TypeError("Unknown byte specification {}".format(
                                             str(byte_format_specification)))
 
-def genCubes(layer_outline, spacer):
-    layer_cubed = np.array([cube2(x, spacer)
+def genCubes(layer_outline, spacer, dims):
+    layer_cubed = np.array([cube2(x, spacer, dims)
                                     for x in layer_outline]).flatten()
     return layer_cubed, len(layer_cubed)/3
 
-def cube(vec, spacer=0.1):
+def subsample(xc, yc, zc, subsample=2):
+    xskip = 0
+    yskip = 0
+    zskip = 0
+    index_mask = []
+    layer_skip = None
+    list_length = xc*yc*zc
+    for i in range(list_length):
+        if (not xskip%subsample) and (not yskip%subsample) and (not zskip%subsample):
+            index_mask.append(i)
+        xskip+=1
+        if xskip%xc == 0:
+            yskip+=1
+            xskip = 0
+            if yskip%yc == 0:           
+                if layer_skip is None:
+                    layer_skip = i     
+                zskip+=1
+                yskip = 0
+    if layer_skip is None:
+        layer_skip = i
+    return np.array(index_mask, dtype=np.int), layer_skip
+
+def cube(vec, spacer=0.1, dims=(0.1, 0.1, 0.1)):
```

Comment:
Dims variable not used. Maybe you forgot to change spacer? Or is it just passed to have same arguments as cube2?

Reply:
actually, spacer variable is not used. But good point, forgot to refactor so that spacer is removed